File: src_java/gestorAplicacion/boleteria/Pelicula.java
//Funcionalidad de la clase: Este contiene anadir la cantidad de Boletos, en este se guardan los atributos de la pelicula 
// entre los que se encuentran nombre, genero, duracion, lenguaje y clasificacion

//Autores: Daniel Santiago Cadavid, Marlon Calle, Daniel Daza, Juan Esteban Ochoa


package gestorAplicacion.boleteria;
import java.io.Serializable;
import java.util.*;
import gestorAplicacion.cinema.Cine;
//import gestorAplicacion.cinema.Cliente;

public class Pelicula implements Serializable{
	private static final long serialVersionUID = 1L;
	private String nombre;
	private String genero;
	private int duracion;
	private String lenguaje;
	private int clasificacion;
	private int cantidadTotalBoletosVendidos;
	private int expectativaVentas;
	private Cine cine;
	
	//Constructores
	public Pelicula(String nombre, String genero, int duracion, String lenguaje, int clasificacion,Cine cine) {
		this.nombre=nombre;
		this.genero=genero;
		this.duracion=duracion;
		this.lenguaje=lenguaje;
		this.clasificacion=clasificacion;
		this.cine=cine;
		cine.agregarPelicula(this);			//Se agrega la pelicula a la lista de peliculas del cine
	}
	
	
	public Pelicula(String nombre, String genero, int duracion, String lenguaje, int clasificacion) {
		super();
		this.nombre = nombre;
		this.genero = genero;
		this.duracion = duracion;
		this.lenguaje = lenguaje;
		this.clasificacion = clasificacion;
	}



	public Pelicula(){
		
	}
	
	//
	//methods
	//


	public void anadirCantidadBoletos(){
		/*No recibe nada y tampoco devuelve nada, este metodo se usa para sumar la cantidad de boletos vendidos		 
		*/
		cantidadTotalBoletosVendidos++;	//Sumar 1 al atributo de la cantidad de total de boletos
		
	}

	
	//
	//getting and setting:
	//
	
	
	public String getNombre() {
		return nombre;
	}
	public void setNombre(String nombre) {
		this.nombre = nombre;
	}
	
	
	public String getGenero() {
		return genero;
	}
	public void setGenero(String genero) {
		this.genero = genero;
	}
	
	
	public int getDuracion() {
		return duracion;
	}
	public void setDuracion(int duracion) {
		this.duracion = duracion;
	}
	
	
	public String getLenguaje() {
		return lenguaje;
	}
	public void setLenguaje(String lenguaje) {
		this.lenguaje = lenguaje;
	}
	
	
	public int getClasificacion() {
		return clasificacion;
	}
	public void setClasificacion(int clasificacion) {
		this.clasificacion = clasificacion;
	}
	
	
	public int getCantidadTotalBoletosVendidos() {
		return cantidadTotalBoletosVendidos;
	}
	public void setCantidadTotalBoletosVendidos(int cantidadTotalBoletosVendidos) {
		this.cantidadTotalBoletosVendidos = cantidadTotalBoletosVendidos;
	}
	
	
	public int getExpectativaVentas() {
		return expectativaVentas;
	}
	public void setExpectativaVentas(int expectativaVentas) {
		this.expectativaVentas = expectativaVentas;
	}

}

File: src/gestionAplicacion/boleteria/pelicula.py
from tokenize import String
from gestionAplicacion.cinemas import Cine

class Pelicula:
    def __init__(self,nombre="",genero="",duracion=0,lenguaje="",clasificacion=0,  cine= None): #!!! pendiente para corregir
        self._nombre:str = nombre
        self._genero:str = genero
        self._duracion:int = duracion
        self._lenguaje:str = lenguaje
        self._clasificacion:int = clasificacion
        self._cine: Cine = cine #!!! pendiente para corregir
        self._cantidadTotalBoletosVendidos:int =0
        self._expectativaVentas:int =0

    def anadirCantidadBoletos(self):
        self._cantidadTotalBoletosVendidos+=1


    def getNombre(self)->str:
        return self._nombre
    def setNombre(self, nombre):
        self._nombre = nombre


    def getGenero(self)->str:
        return self._genero
    def setGenero(self, genero):
        self._genero = genero


    def getDuracion(self)->int:
        return self._duracion
    def setDuracion(self, duracion):
        self._duracion = duracion


    def getLenguaje(self)->str:
        return self._lenguaje
    def setLenguaje(self, lenguaje):
        self._lenguaje = lenguaje


    def getClasificacion(self):
        return self._clasificacion
    def setClasificacion(self, clasificacion):
        self._clasificacion = clasificacion


    def getCine(self)->Cine:
        return self._cine
    def setCine(self, cine):
        self._cine = cine